### PR TITLE
add DPAL header and footer

### PR DESCRIPTION
**Pull request by CodyWMitchell**
[RHCLOUD-29118](https://issues.redhat.com/browse/RHCLOUD-29118)

Added header and footer according to [DPAL documentation](https://source.redhat.com/groups/public/itmarketing/it_marketing_wiki/dpal_dynamic_property_auto_loader__ddo_digitaldata_object) to enable Adobe Launch for analytics.

Using the environment variable `REACT_APP_ENV` to change url in script for stage/prod environment

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -50,6 +51,11 @@
       <script src="https://developers.redhat.com/modules/contrib/red_hat_universal_navigation/dist/%40chapeaux/cpx-user/cpx-user.js?rrg73h"></script>
       <script src="https://developers.redhat.com/modules/contrib/red_hat_universal_navigation/js/rh-account-dropdown.js?rrg73h"></script>
       <script src="https://developers.redhat.com/modules/contrib/red_hat_universal_navigation/dist/%40cpelements/rh-site-switcher/dist/site-switcher.umd.min.js?rrg73h"></script>
+      <script type="text/javascript">
+        if (("undefined" !== typeof _satellite) && ("function" === typeof _satellite.pageBottom)) {
+            _satellite.pageBottom();
+        }
+      </script>
```
Comment (josejulio) [inline public/index.html:58]: shouldn't this be in the body? i.e. outside the div?

> Add the following markup as the very last element in your HTML <body> element:

File: public/index.html [hunk 2]
```diff
@@ -8,6 +8,7 @@
     <meta name="theme-color" content="#000000" />
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
 
+    <script id="dpal" src="https://www.redhat.com/ma/dpal.js" type="text/javascript"></script>
```
Comment (josejulio) [inline public/index.html:11]: Do we need to deal with the staging url? i.e. using the other url when not on prod?

Comment (CodyWMitchell) [reply public/index.html]: Can we use %NODE_ENV% here to check the environment?

File: public/index.html (hunk 1 above)
Comment (CodyWMitchell) [reply public/index.html]: updated! :+1:

**Comment by tahmidefaz** [comment]
@CodyWMitchell Is this ready for review now?

**Comment by CodyWMitchell** [comment]
> @CodyWMitchell Is this ready for review now?

@tahmidefaz Still working on it! Should be ready soon.

File: public/index.html [hunk 3]
```diff
@@ -10,13 +10,24 @@
...
-      dpal.src = ("%NODE_ENV%"==="production") ?
-        'https://www.redhat.com/ma/dpal.js' :
-        'https://www.redhat.com/ma/dpal-staging.js';
-      document.getElementsByTagName('head')[0].appendChild(dpal);
+      (function (env) {
+        let dpal = document.createElement('script');
+        dpal.id = 'dpal';
+        dpal.type = 'text/javascript';
+
+        console.log(env);
```
Comment (josejulio) [inline public/index.html]: Could you remove the console.log? I don't think we really need it

File: public/index.html [hunk 4]
```diff
@@ -10,13 +10,24 @@
...
     <script>
-      let dpal = document.createElement('script');
-      dpal.id = 'dpal';
-      dpal.type = 'text/javascript';
-      dpal.src = ("%NODE_ENV%"==="production") ?
-        'https://www.redhat.com/ma/dpal.js' :
-        'https://www.redhat.com/ma/dpal-staging.js';
-      document.getElementsByTagName('head')[0].appendChild(dpal);
+      (function (env) {
+        let dpal = document.createElement('script');
```
Comment (josejulio) [inline public/index.html:14]: nitpick, this can probably be a const instead of a let

**Review by josejulio — APPROVED** [review]
lgtm

**Review by tahmidefaz — APPROVED** [review]

**Pull request merged**